server/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -291,6 +291,20 @@ public class SearchModule {
         Setting.Property.NodeScope
     );
 
+    public static final Setting<List<String>> SCRIPTED_METRICS_AGG_ALLOWED_SCRIPTS_SETTING = Setting.stringListSetting(
+        // TODO: maybe this should rather be "script.allowed_metric_scripts"
+        "search.aggs.allowed_metric_scripts",
+        Setting.Property.NodeScope,
+        // TODO: remove, only for testing
+        Setting.Property.Dynamic
+    // TODO: Needed?
+    // Setting.Property.Final
```

Comment:
I think we don't want this to be final - we want to be able to change it on the fly at least for testing.